src/auth/oauth2client.ts

```diff
@@ -296,6 +296,12 @@ export interface GenerateAuthUrlOpts {
    * must be used with the 'code_challenge' parameter described above.
    */
   code_challenge?: string;
+
+  /**
+   * A way for developers and/or the auth team to provide a set of key value
+   * pairs to be added as query parameters to the authorization url.
+   */
+  [key: string]: any;
```

Comment:
I think we want more specific types here for the index:

```suggestion
  [key: string]: querystring.ParsedUrlQueryInput;
```